mintpy/view.py

```diff
@@ -1572,12 +1580,17 @@ def plot(self):
                 data[data != 0.] -= ref_data
 
             # masking
-            if self.zero_mask:
-                vprint('masking pixels with zero value')
-                data = np.ma.masked_where(data == 0., data)
             if self.msk is not None:
                 vprint('masking data')
                 data = np.ma.masked_where(self.msk == 0., data)
+            if self.msk is None:
+                self.msk = np.ones(data.shape, dtype=np.int8)
+            if self.zero_mask:
+                vprint('masking pixels with zero value')
+                self.msk = np.ma.masked_array(self.msk, mask=np.isnan(data))
+                np.ma.set_fill_value(self.msk, 0)
+                self.msk=self.msk.filled()
+                data = np.ma.masked_where(self.msk == 0., data)
```

Comment:
I don't get the reason behind this change, could you describe?

Reply:
The rationale was to update the mask to reflect 0-data values in the input data, if specified. Makes sense?

Reply:
I meant: is the original code not working? If so, why?

Reply:
Originally, the code was not working as intended -- inline with my commits -- as the mask in memory was not updated to reflect the zero pixels in the input data. My commits access this mask in order to mask the DEM and GPS stations, if specified.

Reply:
not really. The previous version of PR broke the masking of pixels with zero data value. I have reverted it. I have saved the `self.msk` for the purpose of this PR, so it should still work for you.